GitUI/CommandsDialogs/ToolStripPushButton.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Windows.Forms;
+using GitCommands;
+using GitCommands.Git;
+using GitUI.Properties;
+using ResourceManager;
+
+namespace GitUI.CommandsDialogs
+{
+    public class ToolStripPushButton : ToolStripButton
+    {
+        private readonly TranslationString _push = new TranslationString("Push");
+
+        private readonly TranslationString _aheadCommitsToPush =
+            new TranslationString("{0} new commit(s) will be pushed");
+
+        private readonly TranslationString _behindCommitsTointegrateOrForcePush =
+            new TranslationString("{0} commit(s) should be integrated (or will be lost if force pushed)");
+
+        private IAheadBehindDataProvider _aheadBehindDataProvider;
+        private bool _supportsAheadBehindData;
+
+        public void Initialize(IAheadBehindDataProvider aheadBehindDataProvider, bool supportsAheadBehindData)
+        {
+            _aheadBehindDataProvider = aheadBehindDataProvider;
+            _supportsAheadBehindData = supportsAheadBehindData;
+            ResetToDefaultState();
+        }
+
+        public void DisplayAheadBehindInformation(string branchName)
+        {
+            if (!_supportsAheadBehindData)
+            {
+                return;
+            }
+
+            ResetToDefaultState();
+
+            var aheadBehindData = _aheadBehindDataProvider.GetData(branchName);
```

Comment:
$ git for-each-ref --format="%(push:track) %(refname:short)" refs/heads/pmiossec/_display_ahead_behind_push_data 
 pmiossec/_display_ahead_behind_push_data                                                                        

This do obviously not match the regexp and aheadBehindData will be null.

However 
```
$ git for-each-ref --format="%(upstream:track)" refs/heads/pmiossec/_display_ahead_behind_push_data  
[ahead 1]                                                                                            
                                                                                                     
ejgo@ejgo3 MINGW64 /f/dev/gc/gitextensions_4 (pmiossec/_display_ahead_behind_push_data)              
$ git for-each-ref --format="%(push:track)" refs/heads/pmiossec/_display_ahead_behind_push_data      

$
```

Reply:
upstream works for me, but I do not have push specifically set. The proper solution is to get both upstream and push to get the "best" ahead/behind